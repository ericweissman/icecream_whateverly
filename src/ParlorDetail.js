import React, { Component } from 'react';
import './css/Main.scss';

class ParlorDetail extends Component {
    constructor(props) {
        super(props);

    }
    
    render() {
        return (
        <div className="parlor-detail-card">
            <img src={`./images/Parlors/${this.props.parlor.parlorName}.png`}/>
            <a href={this.props.parlor.website} target="_blank">{this.props.parlor.parlorName}</a>
            <p className='italic'>{this.props.parlor.address}</p>
            <p>{this.props.parlor.phoneNumber}</p>
            <ul>
                <li className='bold'>HOURS</li>
                    {
                        this.props.parlor.daysOpen.map((day) => {
                            return (
                                <div>
                                    <li>{day.day}: {day.open} - {day.close}</li>
                                </div>
                            )
                        })
                    }
            </ul>
            <p>Rating: {this.props.parlor.review}</p>
            <p>Number of Flavors: {this.props.parlor.numberOfFlavors}</p>
            <button className="back-to-parlors-btn" onClick={this.props.getParlorDetails}>Back To Parlors</button>
        
        </div>
        )
    }
}
export default ParlorDetail;