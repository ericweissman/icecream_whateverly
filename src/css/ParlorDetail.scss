@import '_variables.scss';

.parlor-detail-card {
  align-items: center;
  background-color: $light-yellow;
  border: 2px solid black;
  bottom: 0;
  display:flex;
  flex-direction: column;
  height: 90%;
  left: 0;
  margin: auto;
  overflow: scroll;
  position: fixed;
  right: 0;
  top: 0;
  width: 90%;
  z-index: 3;
   a {
      color: $turquoise-primary-color;
      font-size: 30px;
    }
    p {
      line-height: 0px;
    }
    img {
      padding-bottom: 25px;
      padding-top: 25px;
      width: 220px;
    }
    ul {
      list-style: none;
      padding: 0;
    }
    li {
      line-height: 100%;
    }
    button {
      height: 30px;
      padding-bottom: 30px;
      width: 150px;
  }
  .bold {
    font-weight: 600;
  }
  .parlor-address {
    font-style: italic;
  }
 }
