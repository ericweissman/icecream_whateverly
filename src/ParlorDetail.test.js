import React from 'react';
import ParlorDetail from './ParlorDetail';
import { shallow } from 'enzyme';

const parlors = [{
    "parlorName": "Little Man Ice Cream",
    "description": "Quirky shop for exotic ice cream",
    "address": "2620 16th St, Denver, CO 80211",
    "daysOpen": [
      {
        "day": "Sunday",
        "open": "11 AM",
        "close": "10 PM"
      },
      {
        "day": "Monday",
        "open": "11 AM",
        "close": "10 PM"
      },
      {
        "day": "Tuesday",
        "open": "11 AM",
        "close": "10 PM"
      },
      {
        "day": "Wednesday",
        "open": "11 AM",
        "close": "10 PM"
      },
      {
        "day": "Thursday",
        "open": "11 AM",
        "close": "10 PM"
      },
      {
        "day": "Friday",
        "open": "11 AM",
        "close": "11 PM"
      },
      {
        "day": "Saturday",
        "open": "11 AM",
        "close": "11 PM"
      }
    ],
    "phoneNumber": "303-455-3811",
    "website": "http://www.littlemanicecream.com/latest-scoop/menu/",
    "flavors": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27],
    "numberOfFlavors": 27,
    "priceRange": "$",
    "review": 4.7
  }]

describe('ParlorDetail', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ParlorDetail parlor={parlor} />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    
    expect(wrapper).toMatchSnapshot();
  });
})