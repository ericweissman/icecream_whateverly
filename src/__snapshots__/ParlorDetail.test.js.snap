// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParlorDetail should match the snapshot with all data passed in correctly 1`] = `undefined`;
